### PR TITLE
[ZEPPELIN-1821] Add HTTP client to elasticsearch interpreter

### PR DESCRIPTION
**Pull request by bbonnin**
### What is this PR for?
Add HTTP client to elasticsearch interpreter.

### What type of PR is it?
Feature

### Todos
* [X] - Source code
* [X] - Tests
* [X] - License
* [X] - Docs

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-1821

### How should this be tested?
* Start an Elasticsearch node
* Configure the elasticsearch interpreter to use http
* Create queries in a note using elasticsearch

### Questions:
* Does the licenses files need update? Yes
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes

### CODE REVIEW

**Comment by 1ambda** [comment]
great Contribution for elasticsearch interpreter! This PR helps us to support ES 5.0 easily!

**Comment by jongyoul** [comment]
@1ambda Can you review this PR deeply?

**Comment by Leemoonsoo** [comment]
@bbonnin Recently we have fixed some flaky test on master branch. Could you try rebase and see if CI becomes green?

**Comment by bbonnin** [comment]
It looks better now, all is green !

**Comment by 1ambda** [comment]
@bbonnin 👍 

@jongyoul I will take a look more deeply :)

**Comment by Leemoonsoo** [comment]
LGTM @1ambda do you have any comment?

**Comment by jaspinderdineout** [comment]
When will this be merged?

**Comment by 1ambda** [comment]
Works well as described.

- tested on elasticsearch 2.4.4 (transport, http) and 5.0.0 (http only)
- code looks nice. well abstracted client classes is shipped with some tests.

there some commented code lines. LGTM except that.

File: elasticsearch/src/main/java/org/apache/zeppelin/elasticsearch/action/ActionResponse.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ */
+public class ActionResponse {
+
+  private boolean succeeded;
+  private long totalHits;
+  private final List<HitWrapper> hits = new LinkedList<>();
+  private final List<AggWrapper> aggregations = new LinkedList<>();
+
+
+//  public ActionResponse source(String source) {
```
Comment (1ambda) [inline elasticsearch/src/main/java/org/apache/zeppelin/elasticsearch/action/ActionResponse.java]: Can we remove this?

File: elasticsearch/src/main/java/org/apache/zeppelin/elasticsearch/action/ActionResponse.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  public ActionResponse succeeded(boolean succeeded) {
+    this.succeeded = succeeded;
+    return this;
+  }
+
+  public boolean isSucceeded() {
+    return succeeded;
+  }
+
+//  public String getSource() {
```
Comment (1ambda) [inline elasticsearch/src/main/java/org/apache/zeppelin/elasticsearch/action/ActionResponse.java]: comment code lines

File: elasticsearch/src/main/java/org/apache/zeppelin/elasticsearch/action/ActionResponse.java (hunk 1 above)
Comment (bbonnin) [inline elasticsearch/src/main/java/org/apache/zeppelin/elasticsearch/action/ActionResponse.java]: I have removed the commed code lines.

**Comment by Leemoonsoo** [comment]
CI failure looks unrelated, since it only removed commented-out code from last CI success.
Thanks @bbonnin for great contribution.
LGTM and merge to master if no further discussions!

**Comment by jaspinderdineout** [comment]
Why is this not merged into apache:master?

**Comment by 1ambda** [comment]
@jaspinderdineout This was merged into master.

**Pull request closed without merging**